torchrl/modules/tensordict_module/world_models.py

```diff
@@ -0,0 +1,56 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+
+from math import sqrt
+
+import torch
+import torch.distributions as d
+import torch.nn as nn
+import torch.nn.functional as F
+
+from torchrl.modules.tensordict_module import TensorDictModule, TensorDictSequence
+
+__all__ = ["WorldModelWrapper"]
+
+
+class WorldModelWrapper(TensorDictSequence):
+    """
+    World model wrapper.
+    This module wraps together a world model and a reward model.
+    The world model is used to predict an imaginary world state.
+    The reward model is used to predict the reward of the imaginary world state.
+
+    Args:
+        world_model (TensorDictModule): a world model that generates a world state.
+        reward_model (TensorDictModule): a reward model, that reads the world state and returns a reward
+
+    """
+
+    def __init__(
+        self,
+        world_model: TensorDictModule,
+        reward_model: TensorDictModule,
+    ):
+        super().__init__(
+            world_model,
+            reward_model,
+        )
+
+    def get_world_modeler_operator(self) -> TensorDictSequence:
+        """
+
+        Returns a stand-alone policy operator that maps an observation to an action.
```

Comment:
Is this a policy operator mapping observations to actions? Shouldn't this be something like mapping a model input (which could include observation, latent state, action, etc.) to a new world model state?

Reply:
Likewise for the reward operator. Maybe this was some actor-critic copy paste? :P

Reply:
No, this is not a mapping of obs to actions its a mapping from world latent state to next world latent state. The action part is provided separatly:
```
world_modeler = TensorDictModule(
                    ActionStateMergeLinear(5, 4),
                    in_keys=["hidden_state", "action"],
                    out_keys=["next_hidden_state"],
                )
reward_model = TensorDictModule(
                   ActionStateMergeLinear(5, 1),
                    in_keys=["hidden_state", "action"]
                    out_keys=["reward"],
                )
world_model = WorldModelWrapper(
                world_modeler,
                reward_model,
            ),
  ```
The description is indeed wrong i will fix it thanks for pointing it out